overlord/hookstate/hookmgr.go

```diff
@@ -89,9 +90,14 @@ func Manager(s *state.State) (*HookManager, error) {
 }
 
 // HookTask returns a task that will run the specified hook.
-func HookTask(s *state.State, taskSummary, snapName string, revision snap.Revision, hookName string) *state.Task {
+func HookTask(s *state.State, taskSummary, snapName string, revision snap.Revision, hookName string, payload interface{}) *state.Task {
```

Comment:
I'm not sure  I understand payload here, I don't understand what's is the relation with Context.Get/Set, are the latter intended only for hook-to-hook state keeping? otoh having just one big payload seems a bit reductive